chatlas/_chat.py

```diff
@@ -823,6 +832,176 @@ async def extract_data_async(
         json = res[0]
         return json.value
 
+    async def _register_mcp_tools(
+        self,
+        session: MCPClientSession,
+        include_tools: Optional[list[str]] = None,
+        exclude_tools: Optional[list[str]] = None,
+    ):
+        assert not (include_tools and exclude_tools), (
+            "Cannot specify both include_tools and exclude_tools."
+        )
+
+        response = await session.list_tools()
+        for tool in response.tools:
+            if include_tools:
+                if tool.name not in include_tools:
+                    continue
+            if exclude_tools:
+                if tool.name in exclude_tools:
+                    continue
+            self._tools[tool.name] = Tool.from_mcp(
+                session=session,
+                mcp_tool=tool,
+            )
+
+    async def register_sse_mcp_server_async(
```

Comment:
I think `register_mcp_*()` works better for autocomplete reasons

Also, if we don't anticipate the need for a sync version, I'm alright with leaving the `_async` off.

```suggestion
    async def register_mcp_sse_server(
```